### ISSUE TITLE
The current app does not comply with Including-Kendo-professional-in-Aurelia

### ISSUE THREAD

**Issue by adriatic** [aurelia-widgets/aurelia-kendoui#5]
Jeroen wrote a nice advice on how to handle the inclusion of the `kendo.all.min.js` file - see [this](https://github.com/aurelia-widgets/aurelia-kendoui/wiki/Including-Kendo-professional-in-Aurelia)

Jedd, can you please make the needed adjustments (I did not want to do it myself being busy with the Bootstrap side of aurelia-widgets)

**Comment by jedd-ahyoung** [comment]
I thought about doing this before, but as the fix is a configuration
override, I'm not sure what the best method is. Comments in config.js will
be stripped by JSPM and if someone does not have Kendo Professional
installed, they won't be able to run the app.

I'm wondering if having documentation is the better solution so that people
can pull the test project and set things up themselves.

**Comment by adriatic** [comment]
> I'm wondering if having documentation is the better solution so that people
can pull the test project and set things up themselves.

At least initially, I would think that this is the best approach. I think that everyone on this team has a copy of the `kendo.all.min.js` file, so please define **where should it be** so it does not get overwritten and and **how does the app get access to it**

**Comment by JeroenVinke** [comment]
Although I'm very low on time, I want to throw this idea out here. We could use kendo-free by default (and have a dependency on it in package.json), and provide a blocking mechanism if people would use our kendo-pro wrappers with this kendo-free version. The blocking mechanism would prevent initialization of the kendo widget and could provide a helpful error message to warn them they are using a component which is only supported in the professional version of kendo.

We could then provide steps in the documentation on how to convert from kendo-free to kendo-pro. so a question which would need to be answered is: can we detect which version of kendo is used (free or pro)? this would be needed for the blocking mechanism above to work. Also, am I overlooking things?

btw.. this blocking mechanism is something which could be used by aurelia-bootstrap as well to allow the developer to only select a few bootstrap components instead of downloading the full version.

**Comment by adriatic** [comment]
Thanks Jeroen for re-surfacing for air, giving us the chance to resolve this.While my ultimate plan is to contact Telerik and ask them for advice how to address this, I would rather wait a little so we can show them something.

Telerik's own solution is quite simple: everyone can get the KendoUI standard freem from Github, while the KendoUI PRO is purchased from Telerik's website - and that version contains min.js files for all controls that are contained in the PRO version. At least until recently Telerik did not have any sophisticated run-time checks (like embedded secret ID that you get when purchasing PRO)

Assuming that this is still the case, I propose that we follow Telerik's model - have two Aurelia plugins, one for KendoUI PRO and the other for the KendoUI standard. This way someone who has not purchased KendoUI PRO from Telerik, but tries to cheat by fetching KendoUI PRO Aurelia plugin will simply find that is does not support the controls that belong to PRO package only

**Comment by JeroenVinke** [comment]
I like it. So one repository would hold our KendoUI standard wrappers, and another one will have the PRO wrappers. I assume if a developer wanted both, they would install our two plugins.

Another thing to discuss is if we're going to add a dependency on KendoUI in the package.json. this would only be possible for the standard version since the PRO version is not on github. If we add the dependency, and then a developer would install our aurelia-kendo-standard plugin it would automatically download and install KendoUI standard (because of the dependency).

However, if you then also wanted to install aurelia-kendo-pro, you would have to install KendoUI PRO manually, and you would then end up with KendoUI Standard (because of the dep in aurelia-kendo-standard) AND Pro (because the developer downloaded this manually). 

I like that aurelia-kendo-standard would install KendoUI standard automatically. Perhaps we can use documentation to have developers override the dependency (to remove it) in the aurelia-kendo-standard plugin so we would not have a conflict in which there is both KendoUI standard and KendoUI Pro loaded.

**Comment by adriatic** [comment]
> Perhaps we can use documentation to have developers override the dependency (to remove it) in the aurelia-kendo-standard plugin so we would not have a conflict in which there is both KendoUI standard and KendoUI Pro loaded.

Perfect - that is the best way as I can see it Jeroen!

**Comment by jedd-ahyoung** [comment]
I've referenced this in https://github.com/aurelia-widgets/aurelia-kendoui/commit/88a24b7c20f0ff5c8361e18b7f354efb7565af7e, but there's a larger issue at play here. To clarify: will we create one library for all components, or will we create a library for kendo-core and a separate one for kendo-professional?

**Comment by adriatic** [comment]
Until / unless we get some additional insight from Telerik (event that should take place once we have something to show them - it would be a lot easier to discuss that then), I believe that having two plugins, one for kendo-core and the other one for kendo professional. 

The issue of upgrading kendo-core to kendo professional is not clear to me. Here is a typical use case scenario:

I grab the **trial** version of kendo professional  from Telerik's site. After using it for a month (or whatever the length of the trial is set to) controls from the pro subset will start rendering themselves with a watermark that says "trial copy" - a behavior that clearly prevents the use of this version in deployed applications. The controls remain otherwise functional for ever.

If this is still the way Telerik protects their intellectual property, having the "right" version of the file `kendo-all-min.js` is the difference between having the pro or core version.

In conclusion, given this difference between core and pro, it is easier for us to stay on "Telerik's good side" if we deliver Aurelia-KendoUI as two plugin (each of them offering the choice between loading all controls at once - or any subset)
